src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1557,56 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        private static long FileTimeToPosix(FILETIME ft)
+        {
+            long date = ((long)ft.dwHighDateTime << 32) + ft.dwLowDateTime;
+            // removes the diff between 1970 and 1601
+            // 100-nanoseconds = milliseconds * 10000
+            date -= 11644473600000L * 10000;
+
+            // converts back from 100-nanoseconds to seconds
+            return date / 10000000;
+        }
+
+        protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArguments, ModulesResponse> responder)
+        {
+            var response = new ModulesResponse();
+
+            IEnumDebugModules2 enumDebugModules;
+            if (m_program.EnumModules(out enumDebugModules) == HRConstants.S_OK)
+            {
+                var debugModules = new IDebugModule2[1];
+
+                uint numReturned = 0;
+                while (enumDebugModules.Next(1, debugModules, ref numReturned) == HRConstants.S_OK && numReturned == 1)
+                {
+                    var debugModuleInfos = new MODULE_INFO[1];
+                    if (debugModules[0].GetInfo(enum_MODULE_INFO_FIELDS.MIF_ALLFIELDS, debugModuleInfos) == HRConstants.S_OK)
+                    {
+                        var debugModuleInfo = debugModuleInfos[0];
+                        var mod = new ProtocolMessages.Module(debugModuleInfo.m_bstrName, debugModuleInfo.m_bstrName)
```

Comment:
Using the name (`debugModuleInfo.m_bstrName`) of the module is not reccomended for the Module ID. 

Create a static int for the count of modules and use that as the ID.

```
private static uint s_nextModuleId = 0;

static public uint GetNextModuleId()
{
   return ++s_nextModuleId;
}
```

Reply:
might want to start something larger than 0 (1000?)